src/systemctl/systemctl-enable.c

```diff
@@ -315,44 +315,66 @@ int verb_enable(int argc, char *argv[], void *userdata) {
+                const char *start_verb;
+
+                if (streq(verb, "enable"))
+                        start_verb = "start";
+                else if (STR_IN_SET(verb, "disable", "mask"))
+                        start_verb = "stop";
+                else if (streq(verb, "reenable"))
+                        /* Note that we use try-restart here. This matches the semantics of reenable better,
+                         * and allows us to glob template units. */
+                        start_verb = "try-restart";
+                else
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                               "--now can only be used with verb enable, disable, or mask.");
+                                               "--now can only be used with verb enable, disable, reenable, or mask.");
 
                 if (install_client_side())
                         return log_error_errno(SYNTHETIC_ERRNO(EREMOTE),
                                                "--now cannot be used when systemd is not running or in conjunction with --root=/--global, refusing.");
 
                 assert(bus);
 
-                if (strv_extend(&new_args, streq(verb, "enable") ? "start" : "stop") < 0)
+                if (strv_extend(&new_args, start_verb) < 0)
                         return log_oom();
 
                 STRV_FOREACH(name, names) {
-                        if (streq(verb, "enable")) {
-                                char *fn;
+                        _cleanup_free_ char *fn = NULL;
+                        const char *unit_name;
 
-                                /* 'enable' accept path to unit files, so extract it first. Don't try to
-                                 * glob them though, as starting globbed unit seldom makes sense and
-                                 * actually changes the semantic (we're operating on DefaultInstance=
-                                 * when enabling). */
+                        if (STR_IN_SET(verb, "enable", "reenable")) {
+                                /* 'enable' and 'reenable' accept path to unit files, so extract it first. */
 
                                 r = path_extract_filename(*name, &fn);
                                 if (r < 0)
                                         return log_error_errno(r, "Failed to extract filename of '%s': %m", *name);
 
-                                r = strv_consume(&new_args, fn);
-                        } else if (unit_name_is_valid(*name, UNIT_NAME_TEMPLATE)) {
+                                unit_name = fn;
+                        } else
+                                unit_name = *name;
+
+                        if (unit_name_is_valid(unit_name, UNIT_NAME_TEMPLATE)) {
                                 char *globbed;
 
-                                r = unit_name_replace_instance_full(*name, "*", /* accept_glob = */ true, &globbed);
+                                if (streq(verb, "enable")) {
```

Comment:
doesn't matter, but I would probably do the string comparisons only once up top, and then set some bool var or so that is named `prohibit_templates` or so that you set for the enable case and then check here.

or something like that.

i.e. it's nice to not repeat string comparisons, and dissecting the strings once at the top.